system/src/system_sleep.cpp

```diff
@@ -81,13 +81,17 @@ network_status_t system_sleep_network_suspend(network_interface_index index) {
         }
 #endif
 #if HAL_PLATFORM_RTL872X
-        // P2 doesn't need to turn off the modem manually
-        system_notify_event(network_status, network_status_off);
-#else
+        if (index == NETWORK_INTERFACE_WIFI_STA) {
```

Comment:
This is a platform-specific hack which should not be here in generic system code. I understand it's kind of already here and the reason for that was to be able to bring up P2 stuff quicker, but I'd prefer that we fixed the underlying problem, which is in RtlNcpNetif/RtlNcpClient with regards to how power on/off requests/events are managed. cc @XuGuohui

Reply:
I agree. I will fix the power leakage issue, along with refactoring this alittle bit.

Reply:
OK, I will close this PR and assign the shortcut ticket to @XuGuohui